diff -r 7baa22e6a6b3 src/share/classes/javax/swing/plaf/basic/BasicTreeUI.java
--- a/src/share/classes/javax/swing/plaf/basic/BasicTreeUI.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/plaf/basic/BasicTreeUI.java	Mon Jun 04 20:03:19 2012 +0100
@@ -2896,6 +2896,7 @@
      * Will toggle the expandedness of a node, as well as potentially
      * incrementing the selection.
      */
+    @SuppressWarnings("serial")
     public class TreeTraverseAction extends AbstractAction {
         /** Determines direction to traverse, 1 means expand, -1 means
           * collapse. */
@@ -2928,6 +2929,7 @@
 
     /** TreePageAction handles page up and page down events.
       */
+    @SuppressWarnings("serial")
     public class TreePageAction extends AbstractAction {
         /** Specifies the direction to adjust the selection by. */
         protected int         direction;
@@ -2963,6 +2965,7 @@
     /** TreeIncrementAction is used to handle up/down actions.  Selection
       * is moved up or down based on direction.
       */
+    @SuppressWarnings("serial")
     public class TreeIncrementAction extends AbstractAction  {
         /** Specifies the direction to adjust the selection by. */
         protected int         direction;
@@ -3000,6 +3003,7 @@
       * Scrolls either the first or last cell to be visible based on
       * direction.
       */
+    @SuppressWarnings("serial")
     public class TreeHomeAction extends AbstractAction {
         protected int            direction;
         /** Set to true if append to selection. */
@@ -3034,6 +3038,7 @@
     /**
       * For the first selected row expandedness will be toggled.
       */
+    @SuppressWarnings("serial")
     public class TreeToggleAction extends AbstractAction {
         public TreeToggleAction(String name) {
         }
@@ -3053,6 +3058,7 @@
     /**
      * ActionListener that invokes cancelEditing when action performed.
      */
+    @SuppressWarnings("serial")
     public class TreeCancelEditingAction extends AbstractAction {
         public TreeCancelEditingAction(String name) {
         }
@@ -3165,6 +3171,9 @@
 
     private static final TransferHandler defaultTransferHandler = new TreeTransferHandler();
 
+
+
+    @SuppressWarnings("serial")
     static class TreeTransferHandler extends TransferHandler implements UIResource, Comparator<TreePath> {
 
         private JTree tree;
