diff -r 7baa22e6a6b3 src/share/classes/javax/swing/text/DefaultStyledDocument.java
--- a/src/share/classes/javax/swing/text/DefaultStyledDocument.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/text/DefaultStyledDocument.java	Mon Jun 04 20:04:08 2012 +0100
@@ -71,6 +71,8 @@
  */
 public class DefaultStyledDocument extends AbstractDocument implements StyledDocument {
 
+    private static final long serialVersionUID = 7691029896427935656L;
+
     /**
      * Constructs a styled document.
      *
