diff -r 7baa22e6a6b3 src/share/classes/javax/swing/text/JTextComponent.java
--- a/src/share/classes/javax/swing/text/JTextComponent.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/text/JTextComponent.java	Mon Jun 04 20:04:10 2012 +0100
@@ -2541,6 +2541,7 @@
      * has been added to the <code>java.beans</code> package.
      * Please see {@link java.beans.XMLEncoder}.
      */
+    @SuppressWarnings("serial")
     public class AccessibleJTextComponent extends AccessibleJComponent
     implements AccessibleText, CaretListener, DocumentListener,
                AccessibleAction, AccessibleEditableText,
@@ -3963,6 +3964,7 @@
      * if the UI hasn't supplied one, which would only happen if someone
      * hasn't subclassed Basic.
      */
+    @SuppressWarnings("serial")
     static class DefaultTransferHandler extends TransferHandler implements
                                         UIResource {
         public void exportToClipboard(JComponent comp, Clipboard clipboard,
@@ -4253,6 +4255,7 @@
      * <p>KeymapActionMap is then able to transate the object passed in
      * to either message the Keymap, or message its default implementation.
      */
+    @SuppressWarnings("serial")
     static class KeymapWrapper extends InputMap {
         static final Object DefaultActionKey = new Object();
 
@@ -4311,6 +4314,7 @@
      * <code>KeymapWrapper.DefaultActionKey</code>, the default action is
      * returned, otherwise if the key is an Action, it is returned.
      */
+    @SuppressWarnings("serial")
     static class KeymapActionMap extends ActionMap {
         private Keymap keymap;
 
@@ -4402,6 +4406,7 @@
      * position.  This is mutable so that the event can be reused
      * since caret events can be fairly high in bandwidth.
      */
+    @SuppressWarnings("serial")
     static class MutableCaretEvent extends CaretEvent implements ChangeListener, FocusListener, MouseListener {
 
         MutableCaretEvent(JTextComponent c) {
@@ -4986,6 +4991,7 @@
     //
     // Caret implementation for editing the composed text.
     //
+    @SuppressWarnings("serial")
     class ComposedTextCaret extends DefaultCaret implements Serializable {
         Color bg;
 
