diff -r 7baa22e6a6b3 src/share/classes/javax/swing/plaf/ActionMapUIResource.java
--- a/src/share/classes/javax/swing/plaf/ActionMapUIResource.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/plaf/ActionMapUIResource.java	Mon Jun 04 20:02:49 2012 +0100
@@ -36,6 +36,9 @@
  * @since 1.3
  */
 public class ActionMapUIResource extends ActionMap implements UIResource {
+
+    private static final long serialVersionUID = 9146014059597025626L;
+
     public ActionMapUIResource() {
     }
 }
