diff -r 134d9f4f9498 src/share/classes/sun/tools/tree/ConvertExpression.java
--- a/src/share/classes/sun/tools/tree/ConvertExpression.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/sun/tools/tree/ConvertExpression.java	Mon Mar 12 23:16:44 2012 +0000
@@ -65,7 +65,7 @@
               case TC_BYTE:     return new ByteExpression(right.where, (byte)value);
               case TC_CHAR:     return new CharExpression(right.where, (char)value);
               case TC_SHORT:    return new ShortExpression(right.where, (short)value);
-              case TC_INT:      return new IntExpression(right.where, (int)value);
+              case TC_INT:      return new IntExpression(right.where, value);
               case TC_LONG:     return new LongExpression(right.where, (long)value);
               case TC_FLOAT:    return new FloatExpression(right.where, (float)value);
               case TC_DOUBLE:   return new DoubleExpression(right.where, (double)value);
