diff -r 134d9f4f9498 src/share/classes/java/awt/image/DataBufferInt.java
--- a/src/share/classes/java/awt/image/DataBufferInt.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/java/awt/image/DataBufferInt.java	Mon Mar 12 23:16:21 2012 +0000
@@ -154,7 +154,7 @@
      */
     public DataBufferInt(int dataArray[][], int size) {
         super(UNTRACKABLE, TYPE_INT, size, dataArray.length);
-        bankdata = (int [][]) dataArray.clone();
+        bankdata = dataArray.clone();
         data = bankdata[0];
     }
 
@@ -179,7 +179,7 @@
      */
     public DataBufferInt(int dataArray[][], int size, int offsets[]) {
         super(UNTRACKABLE, TYPE_INT, size, dataArray.length, offsets);
-        bankdata = (int [][]) dataArray.clone();
+        bankdata = dataArray.clone();
         data = bankdata[0];
     }
 
@@ -226,7 +226,7 @@
      */
     public int[][] getBankData() {
         theTrackable.setUntrackable();
-        return (int [][]) bankdata.clone();
+        return bankdata.clone();
     }
 
     /**
@@ -278,7 +278,7 @@
      * @see #getElem(int, int)
      */
     public void setElem(int bank, int i, int val) {
-        bankdata[bank][i+offsets[bank]] = (int)val;
+        bankdata[bank][i+offsets[bank]] = val;
         theTrackable.markDirty();
     }
 }
