diff -r 1980be18d0f8 src/share/classes/java/awt/image/ImagingOpException.java
--- a/src/share/classes/java/awt/image/ImagingOpException.java	Sun Apr 22 21:22:17 2012 +0100
+++ b/src/share/classes/java/awt/image/ImagingOpException.java	Mon Apr 23 20:48:02 2012 +0100
@@ -33,6 +33,8 @@
  */
 public class ImagingOpException extends java.lang.RuntimeException {
 
+    private static final long serialVersionUID = 8026288481846276658L;
+
     /**
      * Constructs an <code>ImagingOpException</code> object with the
      * specified message.
