diff -r 134d9f4f9498 src/share/classes/javax/print/attribute/standard/MediaName.java
--- a/src/share/classes/javax/print/attribute/standard/MediaName.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/javax/print/attribute/standard/MediaName.java	Mon Mar 12 23:16:31 2012 +0000
@@ -99,7 +99,7 @@
      */
     protected String[] getStringTable()
     {
-        return (String[])myStringTable.clone();
+        return myStringTable.clone();
     }
 
     /**
