diff -r 134d9f4f9498 src/solaris/classes/sun/awt/X11/MotifDnDConstants.java
--- a/src/solaris/classes/sun/awt/X11/MotifDnDConstants.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/solaris/classes/sun/awt/X11/MotifDnDConstants.java	Mon Mar 12 23:16:46 2012 +0000
@@ -163,7 +163,7 @@
         XlibWrapper.XGrabServer(newDisplay);
 
         try {
-            XlibWrapper.XSetCloseDownMode(newDisplay, (int)XConstants.RetainPermanent);
+            XlibWrapper.XSetCloseDownMode(newDisplay, XConstants.RetainPermanent);
 
             XSetWindowAttributes xwa = new XSetWindowAttributes();
 
@@ -435,7 +435,7 @@
 
         if (formats.length > 0) {
             // Make a defensive copy.
-            formats = (long[])formats.clone();
+            formats = formats.clone();
 
             Arrays.sort(formats);
         }
