diff -r 134d9f4f9498 src/share/classes/sun/java2d/opengl/OGLSurfaceData.java
--- a/src/share/classes/sun/java2d/opengl/OGLSurfaceData.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/sun/java2d/opengl/OGLSurfaceData.java	Mon Mar 12 23:16:43 2012 +0000
@@ -181,25 +181,25 @@
     static {
         if (!GraphicsEnvironment.isHeadless()) {
             // fbobject currently enabled by default; use "false" to disable
-            String fbo = (String)java.security.AccessController.doPrivileged(
+            String fbo = java.security.AccessController.doPrivileged(
                 new sun.security.action.GetPropertyAction(
                     "sun.java2d.opengl.fbobject"));
             isFBObjectEnabled = !"false".equals(fbo);
 
             // lcdshader currently enabled by default; use "false" to disable
-            String lcd = (String)java.security.AccessController.doPrivileged(
+            String lcd = java.security.AccessController.doPrivileged(
                 new sun.security.action.GetPropertyAction(
                     "sun.java2d.opengl.lcdshader"));
             isLCDShaderEnabled = !"false".equals(lcd);
 
             // biopshader currently enabled by default; use "false" to disable
-            String biop = (String)java.security.AccessController.doPrivileged(
+            String biop = java.security.AccessController.doPrivileged(
                 new sun.security.action.GetPropertyAction(
                     "sun.java2d.opengl.biopshader"));
             isBIOpShaderEnabled = !"false".equals(biop);
 
             // gradshader currently enabled by default; use "false" to disable
-            String grad = (String)java.security.AccessController.doPrivileged(
+            String grad = java.security.AccessController.doPrivileged(
                 new sun.security.action.GetPropertyAction(
                     "sun.java2d.opengl.gradshader"));
             isGradShaderEnabled = !"false".equals(grad);
