diff -r 1980be18d0f8 src/share/classes/java/awt/image/RasterFormatException.java
--- a/src/share/classes/java/awt/image/RasterFormatException.java	Sun Apr 22 21:22:17 2012 +0100
+++ b/src/share/classes/java/awt/image/RasterFormatException.java	Mon Apr 23 20:48:13 2012 +0100
@@ -32,6 +32,8 @@
  */
 public class RasterFormatException extends java.lang.RuntimeException {
 
+    private static final long serialVersionUID = 96598996116164315L;
+
     /**
      * Constructs a new <code>RasterFormatException</code> with the
      * specified message.
