diff -r 134d9f4f9498 src/share/classes/java/awt/Window.java
--- a/src/share/classes/java/awt/Window.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/java/awt/Window.java	Mon Mar 12 23:16:25 2012 +0000
@@ -1866,7 +1866,7 @@
      * @since 1.4
      */
     public synchronized WindowListener[] getWindowListeners() {
-        return (WindowListener[])(getListeners(WindowListener.class));
+        return getListeners(WindowListener.class);
     }
 
     /**
@@ -1882,7 +1882,7 @@
      * @since 1.4
      */
     public synchronized WindowFocusListener[] getWindowFocusListeners() {
-        return (WindowFocusListener[])(getListeners(WindowFocusListener.class));
+        return getListeners(WindowFocusListener.class);
     }
 
     /**
@@ -1898,7 +1898,7 @@
      * @since 1.4
      */
     public synchronized WindowStateListener[] getWindowStateListeners() {
-        return (WindowStateListener[])(getListeners(WindowStateListener.class));
+        return getListeners(WindowStateListener.class);
     }
 
 
@@ -1985,6 +1985,7 @@
         return super.eventEnabled(e);
     }
 
+    @SuppressWarnings("fallthrough")
     /**
      * Processes events on this window. If the event is an
      * <code>WindowEvent</code>, it invokes the
@@ -2014,6 +2015,7 @@
                     break;
                 case WindowEvent.WINDOW_STATE_CHANGED:
                     processWindowStateEvent((WindowEvent)e);
+                
                 default:
                     break;
             }
@@ -2022,6 +2024,7 @@
         super.processEvent(e);
     }
 
+    @SuppressWarnings("fallthrough")
     /**
      * Processes window events occurring on this window by
      * dispatching them to any registered WindowListener objects.
