diff -r 134d9f4f9498 src/solaris/classes/sun/awt/X11/XPopupMenuPeer.java
--- a/src/solaris/classes/sun/awt/X11/XPopupMenuPeer.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/solaris/classes/sun/awt/X11/XPopupMenuPeer.java	Mon Mar 12 23:16:54 2012 +0000
@@ -388,7 +388,7 @@
         if (isEventDisabled(xev)) {
             return;
         }
-        final Component currentSource = (Component)getEventSource();
+        final Component currentSource = getEventSource();
         handleKeyPress(xkey);
     }
 
