diff -r 1980be18d0f8 src/share/classes/java/text/DateFormat.java
--- a/src/share/classes/java/text/DateFormat.java	Sun Apr 22 21:22:17 2012 +0100
+++ b/src/share/classes/java/text/DateFormat.java	Tue Apr 24 02:38:19 2012 +0100
@@ -798,7 +798,7 @@
         private static final long serialVersionUID = 7441350119349544720L;
 
         // table of all instances in this class, used by readResolve
-        private static final Map instanceMap = new HashMap(18);
+        private static final Map<String, Field> instanceMap = new HashMap<String, Field>(18);
         // Maps from Calendar constant (such as Calendar.ERA) to Field
         // constant (such as Field.ERA).
         private static final Field[] calendarToFieldMapping =
