diff -r 7baa22e6a6b3 src/share/classes/javax/swing/JColorChooser.java
--- a/src/share/classes/javax/swing/JColorChooser.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/JColorChooser.java	Mon Jun 04 20:01:35 2012 +0100
@@ -85,6 +85,8 @@
  */
 public class JColorChooser extends JComponent implements Accessible {
 
+    private static final long serialVersionUID = 9168066781620640889L;
+
     /**
      * @see #getUIClassID
      * @see #readObject
@@ -581,6 +583,7 @@
      * Java Accessibility API appropriate to color chooser user-interface
      * elements.
      */
+    @SuppressWarnings("serial")
     protected class AccessibleJColorChooser extends AccessibleJComponent {
 
         /**
@@ -604,6 +607,7 @@
  *
  * Note: This needs to be fixed to deal with localization!
  */
+@SuppressWarnings("serial")
 class ColorChooserDialog extends JDialog {
     private Color initialColor;
     private JColorChooser chooserPane;
@@ -724,6 +728,7 @@
         chooserPane.setColor(initialColor);
     }
 
+    @SuppressWarnings("serial")
     class Closer extends WindowAdapter implements Serializable{
         public void windowClosing(WindowEvent e) {
             cancelButton.doClick(0);
@@ -732,6 +737,7 @@
         }
     }
 
+    @SuppressWarnings("serial")
     static class DisposeOnClose extends ComponentAdapter implements Serializable{
         public void componentHidden(ComponentEvent e) {
             Window w = (Window)e.getComponent();
@@ -741,6 +747,7 @@
 
 }
 
+@SuppressWarnings("serial")
 class ColorTracker implements ActionListener, Serializable {
     JColorChooser chooser;
     Color color;
