diff -r 7baa22e6a6b3 src/share/classes/javax/swing/JSpinner.java
--- a/src/share/classes/javax/swing/JSpinner.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/JSpinner.java	Mon Jun 04 20:01:53 2012 +0100
@@ -125,6 +125,9 @@
  */
 public class JSpinner extends JComponent implements Accessible
 {
+
+    private static final long serialVersionUID = -2038084850119252399L;
+
     /**
      * @see #getUIClassID
      * @see #readObject
@@ -380,6 +383,7 @@
      * We pass <code>Change</code> events along to the listeners with the
      * the slider (instead of the model itself) as the event source.
      */
+    @SuppressWarnings("serial")
     private class ModelListener implements ChangeListener, Serializable {
         public void stateChanged(ChangeEvent e) {
             fireStateChanged();
@@ -905,6 +910,7 @@
      * This subclass of javax.swing.DateFormatter maps the minimum/maximum
      * properties to te start/end properties of a SpinnerDateModel.
      */
+    @SuppressWarnings("serial")
     private static class DateEditorFormatter extends DateFormatter {
         private final SpinnerDateModel model;
 
@@ -1081,6 +1088,7 @@
      * properties to a SpinnerNumberModel and initializes the valueClass
      * of the NumberFormatter to match the type of the initial models value.
      */
+    @SuppressWarnings("serial")
     private static class NumberEditorFormatter extends NumberFormatter {
         private final SpinnerNumberModel model;
 
@@ -1306,6 +1316,7 @@
          * user is inserting text at the end of the document. Completion
          * is done by way of the SpinnerListModel method findNextMatch.
          */
+        @SuppressWarnings("serial")
         private class ListFormatter extends
                           JFormattedTextField.AbstractFormatter {
             private DocumentFilter filter;
