diff -r 7baa22e6a6b3 src/share/classes/javax/swing/JTable.java
--- a/src/share/classes/javax/swing/JTable.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/JTable.java	Mon Jun 04 20:01:54 2012 +0100
@@ -221,6 +221,9 @@
     TableColumnModelListener, ListSelectionListener, CellEditorListener,
     Accessible, RowSorterListener
 {
+
+    private static final long serialVersionUID = -8738988034840546725L;
+
 //
 // Static Constants
 //
@@ -5351,6 +5354,7 @@
     /**
      * Default Renderers
      **/
+    @SuppressWarnings("serial")
     static class NumberRenderer extends DefaultTableCellRenderer.UIResource {
         public NumberRenderer() {
             super();
@@ -5358,6 +5362,7 @@
         }
     }
 
+    @SuppressWarnings("serial")
     static class DoubleRenderer extends NumberRenderer {
         NumberFormat formatter;
         public DoubleRenderer() { super(); }
@@ -5370,6 +5375,7 @@
         }
     }
 
+    @SuppressWarnings("serial")
     static class DateRenderer extends DefaultTableCellRenderer.UIResource {
         DateFormat formatter;
         public DateRenderer() { super(); }
@@ -5382,6 +5388,7 @@
         }
     }
 
+    @SuppressWarnings("serial")
     static class IconRenderer extends DefaultTableCellRenderer.UIResource {
         public IconRenderer() {
             super();
@@ -5390,7 +5397,7 @@
         public void setValue(Object value) { setIcon((value instanceof Icon) ? (Icon)value : null); }
     }
 
-
+    @SuppressWarnings("serial")
     static class BooleanRenderer extends JCheckBox implements TableCellRenderer, UIResource
     {
         private static final Border noFocusBorder = new EmptyBorder(1, 1, 1, 1);
@@ -5447,6 +5454,7 @@
     /**
      * Default Editors
      */
+    @SuppressWarnings("serial")
     static class GenericEditor extends DefaultCellEditor {
 
         Class[] argTypes = new Class[]{String.class};
@@ -5510,6 +5518,7 @@
         }
     }
 
+    @SuppressWarnings("serial")
     static class NumberEditor extends GenericEditor {
 
         public NumberEditor() {
@@ -5517,6 +5526,7 @@
         }
     }
 
+    @SuppressWarnings("serial")
     static class BooleanEditor extends DefaultCellEditor {
         public BooleanEditor() {
             super(new JCheckBox());
@@ -6583,6 +6593,7 @@
      * has been added to the <code>java.beans</code> package.
      * Please see {@link java.beans.XMLEncoder}.
      */
+    @SuppressWarnings("serial")
     protected class AccessibleJTable extends AccessibleJComponent
     implements AccessibleSelection, ListSelectionListener, TableModelListener,
     TableColumnModelListener, CellEditorListener, PropertyChangeListener,
