diff -r 7baa22e6a6b3 src/share/classes/javax/swing/JTextField.java
--- a/src/share/classes/javax/swing/JTextField.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/JTextField.java	Mon Jun 04 20:02:04 2012 +0100
@@ -163,6 +163,8 @@
  */
 public class JTextField extends JTextComponent implements SwingConstants {
 
+    private static final long serialVersionUID = -3082247762805483391L;
+
     /**
      * Constructs a new <code>TextField</code>.  A default model is created,
      * the initial string is <code>null</code>,
@@ -681,6 +683,7 @@
         return new TextFieldActionPropertyChangeListener(this, a);
     }
 
+    @SuppressWarnings("serial")
     private static class TextFieldActionPropertyChangeListener extends
                          ActionPropertyChangeListener<JTextField> {
         TextFieldActionPropertyChangeListener(JTextField tf, Action a) {
