diff -r 7baa22e6a6b3 src/share/classes/javax/swing/SizeRequirements.java
--- a/src/share/classes/javax/swing/SizeRequirements.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/SizeRequirements.java	Mon Jun 04 20:02:10 2012 +0100
@@ -95,6 +95,8 @@
  */
 public class SizeRequirements implements Serializable {
 
+    private static final long serialVersionUID = 9217749429906736553L;
+
     /**
      * The minimum size required.
      * For a component <code>comp</code>, this should be equal to either
