diff -r 7baa22e6a6b3 src/share/classes/javax/swing/colorchooser/ColorPanel.java
--- a/src/share/classes/javax/swing/colorchooser/ColorPanel.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/colorchooser/ColorPanel.java	Mon Jun 04 20:02:31 2012 +0100
@@ -40,6 +40,8 @@
 
 final class ColorPanel extends JPanel implements ActionListener {
 
+    private static final long serialVersionUID = 4699272233105798183L;
+
     private final SlidingSpinner[] spinners = new SlidingSpinner[5];
     private final float[] values = new float[this.spinners.length];
 
