diff -r f9fb8c4b4550 src/share/classes/javax/print/attribute/HashAttributeSet.java
--- a/src/share/classes/javax/print/attribute/HashAttributeSet.java	Mon Jan 30 11:44:45 2012 +0000
+++ b/src/share/classes/javax/print/attribute/HashAttributeSet.java	Sat Feb 11 19:18:52 2012 +0000
@@ -382,7 +382,7 @@
         return
            attribute != null &&
            attribute instanceof Attribute &&
-           attribute.equals(attrMap.get(((Attribute)attribute).getCategory()));
+           attribute.equals(attrMap.get(attribute.getCategory()));
     }
 
     /**
