diff -r 0678af55d3db src/share/classes/java/nio/X-Buffer.java.template
--- a/src/share/classes/java/nio/X-Buffer.java.template	Tue Jun 05 17:11:26 2012 +0800
+++ b/src/share/classes/java/nio/X-Buffer.java.template	Tue Jun 05 19:59:07 2012 +0100
@@ -1046,7 +1046,7 @@
      * @return  A summary string
      */
     public String toString() {
-        StringBuffer sb = new StringBuffer();
+        StringBuilder sb = new StringBuilder();
         sb.append(getClass().getName());
         sb.append("[pos=");
         sb.append(position());
