diff -r 7baa22e6a6b3 src/share/classes/javax/swing/text/html/CSS.java
--- a/src/share/classes/javax/swing/text/html/CSS.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/text/html/CSS.java	Mon Jun 04 20:04:16 2012 +0100
@@ -125,6 +125,8 @@
  */
 public class CSS implements Serializable {
 
+    private static final long serialVersionUID = 3482914137475155648L;
+
     /**
      * Definitions to be used as a key on AttributeSet's
      * that might hold CSS attributes.  Since this is a
