diff -r 7baa22e6a6b3 src/share/classes/javax/swing/text/html/HTMLEditorKit.java
--- a/src/share/classes/javax/swing/text/html/HTMLEditorKit.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/text/html/HTMLEditorKit.java	Mon Jun 04 20:04:21 2012 +0100
@@ -163,6 +163,8 @@
  */
 public class HTMLEditorKit extends StyledEditorKit implements Accessible {
 
+    private static final long serialVersionUID = 2010423999419099584L;
+
     private JEditorPane theEditor;
 
     /**
@@ -641,6 +643,7 @@
      * Class to watch the associated component and fire
      * hyperlink events on it when appropriate.
      */
+    @SuppressWarnings("serial")
     public static class LinkController extends MouseAdapter implements MouseMotionListener, Serializable {
         private Element curElem = null;
         /**
@@ -1593,6 +1596,7 @@
      * addTag. These will be checked for if there is no parentTag at
      * offset.
      */
+    @SuppressWarnings("serial")
     public static class InsertHTMLTextAction extends HTMLTextAction {
         public InsertHTMLTextAction(String name, String html,
                                     HTML.Tag parentTag, HTML.Tag addTag) {
@@ -1824,6 +1828,7 @@
      * the parent HTML.Tag based on the paragraph element at the selection
      * start.
      */
+    @SuppressWarnings("serial")
     static class InsertHRAction extends InsertHTMLTextAction {
         InsertHRAction() {
             super("InsertHR", "<hr>", null, HTML.Tag.IMPLIED, null, null,
@@ -1878,6 +1883,7 @@
      * javax.accessibility package.  The text package should support
      * keyboard navigation of text elements directly.
      */
+    @SuppressWarnings("serial")
     static class NavigateLinkAction extends TextAction implements CaretListener {
 
         private static final FocusHighlightPainter focusPainter =
@@ -2087,6 +2093,7 @@
      * javax.accessibility package.  The text package should support
      * keyboard navigation of text elements directly.
      */
+    @SuppressWarnings("serial")
     static class ActivateLinkAction extends TextAction {
 
         /**
@@ -2260,7 +2267,7 @@
      * @see DefaultEditorKit#beginAction
      * @see HTMLEditorKit#getActions
      */
-
+    @SuppressWarnings("serial")
     static class BeginAction extends TextAction {
 
         /* Create this object with the appropriate identifier. */
