diff -r 7baa22e6a6b3 src/share/classes/javax/swing/plaf/metal/MetalBorders.java
--- a/src/share/classes/javax/swing/plaf/metal/MetalBorders.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/plaf/metal/MetalBorders.java	Mon Jun 04 20:03:21 2012 +0100
@@ -57,6 +57,8 @@
         new StringUIClientPropertyKey("NoButtonRollover");
 
 
+
+    @SuppressWarnings("serial")
     public static class Flush3DBorder extends AbstractBorder implements UIResource{
         public void paintBorder(Component c, Graphics g, int x, int y,
                           int w, int h) {
@@ -73,6 +75,9 @@
         }
     }
 
+
+
+    @SuppressWarnings("serial")
     public static class ButtonBorder extends AbstractBorder implements UIResource {
 
         protected static Insets borderInsets = new Insets( 3, 3, 3, 3 );
@@ -187,6 +192,9 @@
         }
     }
 
+
+
+    @SuppressWarnings("serial")
     public static class InternalFrameBorder extends AbstractBorder implements UIResource {
         private static final int corner = 14;
 
@@ -248,6 +256,7 @@
      * Border for a Frame.
      * @since 1.4
      */
+    @SuppressWarnings("serial")
     static class FrameBorder extends AbstractBorder implements UIResource {
         private static final int corner = 14;
 
@@ -310,6 +319,7 @@
      * Border for a Frame.
      * @since 1.4
      */
+    @SuppressWarnings("serial")
     static class DialogBorder extends AbstractBorder implements UIResource
     {
         private static final int corner = 14;
@@ -403,6 +413,7 @@
      * Border for an Error Dialog.
      * @since 1.4
      */
+    @SuppressWarnings("serial")
     static class ErrorDialogBorder extends DialogBorder implements UIResource
     {
         protected Color getActiveBackground() {
@@ -416,6 +427,7 @@
      * JColorChooser..
      * @since 1.4
      */
+    @SuppressWarnings("serial")
     static class QuestionDialogBorder extends DialogBorder implements UIResource
     {
         protected Color getActiveBackground() {
@@ -428,6 +440,7 @@
      * Border for a Warning Dialog.
      * @since 1.4
      */
+    @SuppressWarnings("serial")
     static class WarningDialogBorder extends DialogBorder implements UIResource
     {
         protected Color getActiveBackground() {
@@ -440,6 +453,7 @@
      * Border for a Palette.
      * @since 1.3
      */
+    @SuppressWarnings("serial")
     public static class PaletteBorder extends AbstractBorder implements UIResource {
         int titleHeight = 0;
 
@@ -462,6 +476,9 @@
         }
     }
 
+
+
+    @SuppressWarnings("serial")
     public static class OptionDialogBorder extends AbstractBorder implements UIResource {
         int titleHeight = 0;
 
@@ -524,6 +541,8 @@
     }
 
 
+
+    @SuppressWarnings("serial")
     public static class MenuBarBorder extends AbstractBorder implements UIResource {
         protected static Insets borderInsets = new Insets( 1, 0, 1, 0 );
 
@@ -560,6 +579,9 @@
         }
     }
 
+
+
+    @SuppressWarnings("serial")
     public static class MenuItemBorder extends AbstractBorder implements UIResource {
         protected static Insets borderInsets = new Insets( 2, 2, 2, 2 );
 
@@ -607,6 +629,9 @@
         }
     }
 
+
+
+    @SuppressWarnings("serial")
     public static class PopupMenuBorder extends AbstractBorder implements UIResource {
         protected static Insets borderInsets = new Insets( 3, 1, 2, 1 );
 
@@ -632,6 +657,8 @@
     }
 
 
+
+    @SuppressWarnings("serial")
     public static class RolloverButtonBorder extends ButtonBorder {
 
         public void paintBorder( Component c, Graphics g, int x, int y, int w, int h ) {
@@ -652,6 +679,7 @@
      * Note: This is identical to the package private class
      * BasicBorders.RolloverMarginBorder and should probably be consolidated.
      */
+    @SuppressWarnings("serial")
     static class RolloverMarginBorder extends EmptyBorder {
 
         public RolloverMarginBorder() {
@@ -681,6 +709,9 @@
         }
     }
 
+
+
+    @SuppressWarnings("serial")
     public static class ToolBarBorder extends AbstractBorder implements UIResource, SwingConstants
     {
         protected MetalBumps bumps = new MetalBumps( 10, 10,
@@ -808,6 +839,9 @@
         return textFieldBorder;
     }
 
+
+
+    @SuppressWarnings("serial")
     public static class TextFieldBorder extends Flush3DBorder {
 
         public void paintBorder(Component c, Graphics g, int x, int y,
@@ -832,6 +866,9 @@
         }
     }
 
+
+
+    @SuppressWarnings("serial")
     public static class ScrollPaneBorder extends AbstractBorder implements UIResource {
         public void paintBorder(Component c, Graphics g, int x, int y,
                           int w, int h) {
@@ -892,6 +929,7 @@
     /**
      * @since 1.3
      */
+    @SuppressWarnings("serial")
     public static class ToggleButtonBorder extends ButtonBorder {
         public void paintBorder(Component c, Graphics g, int x, int y, int w, int h) {
             AbstractButton button = (AbstractButton)c;
@@ -924,6 +962,7 @@
      * Border for a Table Header
      * @since 1.3
      */
+    @SuppressWarnings("serial")
     public static class TableHeaderBorder extends javax.swing.border.AbstractBorder {
         protected Insets editorBorderInsets = new Insets( 2, 2, 2, 0 );
 
