diff -r 134d9f4f9498 src/share/classes/java/awt/image/MultiPixelPackedSampleModel.java
--- a/src/share/classes/java/awt/image/MultiPixelPackedSampleModel.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/java/awt/image/MultiPixelPackedSampleModel.java	Mon Mar 12 23:16:22 2012 +0000
@@ -198,7 +198,7 @@
     public DataBuffer createDataBuffer() {
         DataBuffer dataBuffer = null;
 
-        int size = (int)scanlineStride*height;
+        int size = scanlineStride*height;
         switch (dataType) {
         case DataBuffer.TYPE_BYTE:
             dataBuffer = new DataBufferByte(size+(dataBitOffset+7)/8);
