diff -r 0678af55d3db src/share/classes/javax/sql/rowset/serial/SQLOutputImpl.java
--- a/src/share/classes/javax/sql/rowset/serial/SQLOutputImpl.java	Tue Jun 05 17:11:26 2012 +0800
+++ b/src/share/classes/javax/sql/rowset/serial/SQLOutputImpl.java	Tue Jun 05 19:59:24 2012 +0100
@@ -326,7 +326,7 @@
              int i;
              while( (i = bufReader.read()) != -1 ) {
                 char ch = (char)i;
-                StringBuffer strBuf = new StringBuffer();
+                StringBuilder strBuf = new StringBuilder();
                 strBuf.append(ch);
 
                 String str = new String(strBuf);
@@ -357,7 +357,7 @@
                while( (i=bufReader.read()) != -1 ) {
                 char ch = (char)i;
 
-                StringBuffer strBuf = new StringBuffer();
+                StringBuilder strBuf = new StringBuilder();
                 strBuf.append(ch);
 
                 String str = new String(strBuf);
@@ -387,7 +387,7 @@
              while( (i=bufReader.read()) != -1 ) {
                 char ch = (char)i;
 
-                StringBuffer strBuf = new StringBuffer();
+                StringBuilder strBuf = new StringBuilder();
                 strBuf.append(ch);
 
                 String str = new String(strBuf);
