diff -r 134d9f4f9498 src/share/classes/javax/accessibility/AccessibleStateSet.java
--- a/src/share/classes/javax/accessibility/AccessibleStateSet.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/javax/accessibility/AccessibleStateSet.java	Mon Mar 12 23:16:28 2012 +0000
@@ -171,7 +171,7 @@
         } else {
             AccessibleState[] stateArray = new AccessibleState[states.size()];
             for (int i = 0; i < stateArray.length; i++) {
-                stateArray[i] = (AccessibleState) states.elementAt(i);
+                stateArray[i] = states.elementAt(i);
             }
             return stateArray;
         }
@@ -187,11 +187,9 @@
     public String toString() {
         String ret = null;
         if ((states != null) && (states.size() > 0)) {
-            ret = ((AccessibleState) (states.elementAt(0))).toDisplayString();
+            ret = states.elementAt(0).toDisplayString();
             for (int i = 1; i < states.size(); i++) {
-                ret = ret + ","
-                        + ((AccessibleState) (states.elementAt(i))).
-                                              toDisplayString();
+                ret = ret + "," + states.elementAt(i).toDisplayString();
             }
         }
         return ret;
