diff -r 134d9f4f9498 src/share/classes/com/sun/imageio/plugins/bmp/BMPImageReader.java
--- a/src/share/classes/com/sun/imageio/plugins/bmp/BMPImageReader.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/com/sun/imageio/plugins/bmp/BMPImageReader.java	Mon Mar 12 23:16:06 2012 +0000
@@ -690,6 +690,7 @@
         return metadata.compression == BI_RGB;
     }
 
+    @SuppressWarnings("fallthrough")
     public BufferedImage read(int imageIndex, ImageReadParam param)
         throws IOException {
 
@@ -767,14 +768,11 @@
         // the sampleModel can be null in case of embedded image
         if (sampleModel != null) {
             if (sampleModel.getDataType() == DataBuffer.TYPE_BYTE)
-                bdata = (byte[])
-                    ((DataBufferByte)raster.getDataBuffer()).getData();
+                bdata = ((DataBufferByte)raster.getDataBuffer()).getData();
             else if (sampleModel.getDataType() == DataBuffer.TYPE_USHORT)
-                sdata = (short[])
-                    ((DataBufferUShort)raster.getDataBuffer()).getData();
+                sdata = ((DataBufferUShort)raster.getDataBuffer()).getData();
             else if (sampleModel.getDataType() == DataBuffer.TYPE_INT)
-                idata = (int[])
-                    ((DataBufferInt)raster.getDataBuffer()).getData();
+                idata = ((DataBufferInt)raster.getDataBuffer()).getData();
         }
 
         // There should only be one tile.
