diff -r 7baa22e6a6b3 src/share/classes/javax/swing/text/AbstractDocument.java
--- a/src/share/classes/javax/swing/text/AbstractDocument.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/text/AbstractDocument.java	Mon Jun 04 20:04:02 2012 +0100
@@ -2249,6 +2249,7 @@
      * has been added to the <code>java.beans</code> package.
      * Please see {@link java.beans.XMLEncoder}.
      */
+    @SuppressWarnings("serial")
     public class BranchElement extends AbstractElement {
 
         /**
@@ -2505,6 +2506,7 @@
      *
      * @see     Element
      */
+    @SuppressWarnings("serial")
     public class LeafElement extends AbstractElement {
 
         /**
@@ -2666,6 +2668,7 @@
      * The root element is the only element in the bidi element structure
      * which contains children.
      */
+    @SuppressWarnings("serial")
     class BidiRootElement extends BranchElement {
 
         BidiRootElement() {
@@ -2684,6 +2687,7 @@
     /**
      * Represents an element of the bidirectional element structure.
      */
+    @SuppressWarnings("serial")
     class BidiElement extends LeafElement {
 
         /**
@@ -2724,6 +2728,7 @@
      * This is used by the AbstractDocument class and its extensions
      * for broadcasting change information to the document listeners.
      */
+    @SuppressWarnings("serial")
     public class DefaultDocumentEvent extends CompoundEdit implements DocumentEvent {
 
         /**
@@ -3014,6 +3019,7 @@
      * An implementation of ElementChange that can be added to the document
      * event.
      */
+    @SuppressWarnings("serial")
     public static class ElementEdit extends AbstractUndoableEdit implements DocumentEvent.ElementChange {
 
         /**
