diff -r f9fb8c4b4550 src/share/classes/org/jcp/xml/dsig/internal/dom/DOMXMLSignature.java
--- a/src/share/classes/org/jcp/xml/dsig/internal/dom/DOMXMLSignature.java	Mon Jan 30 11:44:45 2012 +0000
+++ b/src/share/classes/org/jcp/xml/dsig/internal/dom/DOMXMLSignature.java	Sat Feb 11 19:19:08 2012 +0000
@@ -501,7 +501,7 @@
         }
 
         public byte[] getValue() {
-            return (value == null) ? null : (byte[]) value.clone();
+            return (value == null) ? null : value.clone();
         }
 
         public boolean validate(XMLValidateContext validateContext)
