diff -r 134d9f4f9498 src/share/classes/com/sun/java/browser/dom/DOMService.java
--- a/src/share/classes/com/sun/java/browser/dom/DOMService.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/com/sun/java/browser/dom/DOMService.java	Mon Mar 12 23:16:09 2012 +0000
@@ -43,7 +43,7 @@
     {
         try
         {
-            String provider = (String) java.security.AccessController.doPrivileged(
+            String provider = java.security.AccessController.doPrivileged(
                    new sun.security.action.GetPropertyAction("com.sun.java.browser.dom.DOMServiceProvider"));
 
             Class clazz = DOMService.class.forName("sun.plugin.dom.DOMService");
