diff -r 134d9f4f9498 src/share/classes/com/sun/imageio/plugins/common/LZWStringTable.java
--- a/src/share/classes/com/sun/imageio/plugins/common/LZWStringTable.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/com/sun/imageio/plugins/common/LZWStringTable.java	Mon Mar 12 23:16:07 2012 +0000
@@ -143,7 +143,7 @@
     }
 
     static public int hash(short index, byte lastbyte) {
-        return ((int)((short)(lastbyte << 8) ^ index) & 0xFFFF) % HASHSIZE;
+        return (((short)(lastbyte << 8) ^ index) & 0xFFFF) % HASHSIZE;
     }
 
     /*
