diff -r 134d9f4f9498 src/share/classes/java/awt/image/ComponentColorModel.java
--- a/src/share/classes/java/awt/image/ComponentColorModel.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/java/awt/image/ComponentColorModel.java	Mon Mar 12 23:16:18 2012 +0000
@@ -1514,13 +1514,13 @@
                 intpixel[0] = (int) (red * factor * ((1<<nBits[0]) - 1) + 0.5f);
             }
             if (nBits[1] == precision) {
-                intpixel[1] = (int)(grn);
+                intpixel[1] = grn;
             }
             else {
                 intpixel[1] = (int) (grn * factor * ((1<<nBits[1]) - 1) + 0.5f);
             }
             if (nBits[2] == precision) {
-                intpixel[2] = (int)(blu);
+                intpixel[2] = blu;
             }
             else {
                 intpixel[2] = (int) (blu * factor * ((1<<nBits[2]) - 1) + 0.5f);
