diff -r 0678af55d3db src/share/classes/java/security/cert/PKIXBuilderParameters.java
--- a/src/share/classes/java/security/cert/PKIXBuilderParameters.java	Tue Jun 05 17:11:26 2012 +0800
+++ b/src/share/classes/java/security/cert/PKIXBuilderParameters.java	Tue Jun 05 19:59:10 2012 +0100
@@ -189,7 +189,7 @@
      * @return a formatted string describing the parameters
      */
     public String toString() {
-        StringBuffer sb = new StringBuffer();
+        StringBuilder sb = new StringBuilder();
         sb.append("[\n");
         sb.append(super.toString());
         sb.append("  Maximum Path Length: " + maxPathLength + "\n");
