diff -r 134d9f4f9498 src/solaris/classes/sun/awt/X11/XDecoratedPeer.java
--- a/src/solaris/classes/sun/awt/X11/XDecoratedPeer.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/solaris/classes/sun/awt/X11/XDecoratedPeer.java	Mon Mar 12 23:16:49 2012 +0000
@@ -307,7 +307,7 @@
                 insets_corrected = true;
                 return;
             }
-            Component t = (Component)target;
+            Component t = target;
             if (getDecorations() == XWindowAttributesData.AWT_DECOR_NONE) {
                 setReparented(true);
                 insets_corrected = true;
@@ -409,7 +409,7 @@
 
     public void handleMoved(WindowDimensions dims) {
         Point loc = dims.getLocation();
-        AWTAccessor.getComponentAccessor().setLocation((Component)target, loc.x, loc.y);
+        AWTAccessor.getComponentAccessor().setLocation(target, loc.x, loc.y);
         postEvent(new ComponentEvent(target, ComponentEvent.COMPONENT_MOVED));
     }
 
@@ -515,8 +515,8 @@
                 // its location changes.
                 Point oldLocation = getLocation();
 
-                Point newLocation = new Point(AWTAccessor.getComponentAccessor().getX((Component)target),
-                                              AWTAccessor.getComponentAccessor().getY((Component)target));
+                Point newLocation = new Point(AWTAccessor.getComponentAccessor().getX(target),
+                                              AWTAccessor.getComponentAccessor().getY(target));
 
                 if (!newLocation.equals(oldLocation)) {
                     handleMoved(newDimensions);
@@ -714,7 +714,7 @@
         updateChildrenSizes();
 
         // Bounds of the window
-        Rectangle targetBounds = AWTAccessor.getComponentAccessor().getBounds((Component)target);
+        Rectangle targetBounds = AWTAccessor.getComponentAccessor().getBounds(target);
 
         Point newLocation = targetBounds.getLocation();
         if (xe.get_send_event() || runningWM == XWM.NO_WM || XWM.isNonReparentingWM()) {
@@ -1040,10 +1040,10 @@
 
     final void dumpTarget() {
         AWTAccessor.ComponentAccessor compAccessor = AWTAccessor.getComponentAccessor();
-        int getWidth = compAccessor.getWidth((Component)target);
-        int getHeight = compAccessor.getHeight((Component)target);
-        int getTargetX = compAccessor.getX((Component)target);
-        int getTargetY = compAccessor.getY((Component)target);
+        int getWidth = compAccessor.getWidth(target);
+        int getHeight = compAccessor.getHeight(target);
+        int getTargetX = compAccessor.getX(target);
+        int getTargetY = compAccessor.getY(target);
         System.err.println(">>> Target: " + getTargetX + ", " + getTargetY + ", " + getWidth + ", " + getHeight);
     }
 
