diff -r 134d9f4f9498 src/solaris/classes/sun/awt/X11/XIconWindow.java
--- a/src/solaris/classes/sun/awt/X11/XIconWindow.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/solaris/classes/sun/awt/X11/XIconWindow.java	Mon Mar 12 23:16:53 2012 +0000
@@ -292,8 +292,8 @@
             }
             long dst = XlibWrapper.XCreateImage(XToolkit.getDisplay(),
                                                 visInfo.get_visual(),
-                                                (int)awtImage.get_Depth(),
-                                                (int)XConstants.ZPixmap,
+                                                awtImage.get_Depth(),
+                                                XConstants.ZPixmap,
                                                 0,
                                                 bytes,
                                                 iconWidth,
@@ -470,7 +470,7 @@
                 params.add(BACKGROUND_PIXMAP, iconPixmap);
                 params.add(COLORMAP, adata.get_awt_cmap());
                 params.add(DEPTH, awtImage.get_Depth());
-                params.add(VISUAL_CLASS, (int)XConstants.InputOutput);
+                params.add(VISUAL_CLASS, XConstants.InputOutput);
                 params.add(VISUAL, visInfo.get_visual());
                 params.add(VALUE_MASK, XConstants.CWBorderPixel | XConstants.CWColormap | XConstants.CWBackPixmap);
                 params.add(PARENT_WINDOW, XlibWrapper.RootWindow(XToolkit.getDisplay(), visInfo.get_screen()));
