diff -r 134d9f4f9498 src/solaris/classes/sun/awt/X11/XInputMethod.java
--- a/src/solaris/classes/sun/awt/X11/XInputMethod.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/solaris/classes/sun/awt/X11/XInputMethod.java	Mon Mar 12 23:16:53 2012 +0000
@@ -134,7 +134,7 @@
     }
 
     long getCurrentParentWindow() {
-        return (long)((XWindow)clientComponentWindow.getPeer()).getContentWindow();
+        return ((XWindow)clientComponentWindow.getPeer()).getContentWindow();
     }
 
     /*
