diff -r 134d9f4f9498 src/solaris/classes/sun/awt/X11/XTextFieldPeer.java
--- a/src/solaris/classes/sun/awt/X11/XTextFieldPeer.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/solaris/classes/sun/awt/X11/XTextFieldPeer.java	Mon Mar 12 23:16:55 2012 +0000
@@ -75,7 +75,7 @@
         xtext.setCursor(target.getCursor());
         XToolkit.specialPeerMap.put(xtext,this);
 
-        TextField txt = (TextField) target;
+        TextField txt = target;
         initTextField();
         setText(txt.getText());
         if (txt.echoCharIsSet()) {
