diff -r 7baa22e6a6b3 src/share/classes/javax/swing/plaf/BorderUIResource.java
--- a/src/share/classes/javax/swing/plaf/BorderUIResource.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/plaf/BorderUIResource.java	Mon Jun 04 20:02:50 2012 +0100
@@ -61,6 +61,9 @@
  */
 public class BorderUIResource implements Border, UIResource, Serializable
 {
+
+    private static final long serialVersionUID = -3440553684010079691L;
+
     static Border etched;
     static Border loweredBevel;
     static Border raisedBevel;
