diff -r 134d9f4f9498 src/share/classes/java/awt/DefaultKeyboardFocusManager.java
--- a/src/share/classes/java/awt/DefaultKeyboardFocusManager.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/java/awt/DefaultKeyboardFocusManager.java	Mon Mar 12 23:16:17 2012 +0000
@@ -528,7 +528,7 @@
                     // newFocusOwner is not focus traversable.
                     dequeueKeyEvents(-1, newFocusOwner);
                     if (KeyboardFocusManager.isAutoFocusTransferEnabled()) {
-                        restoreFocus(fe, (Window)newFocusedWindow);
+                        restoreFocus(fe, newFocusedWindow);
                     }
                     break;
                 }
@@ -540,7 +540,7 @@
                         // Focus change was rejected. Unlikely, but possible.
                         dequeueKeyEvents(-1, newFocusOwner);
                         if (KeyboardFocusManager.isAutoFocusTransferEnabled()) {
-                            restoreFocus(fe, (Window)newFocusedWindow);
+                            restoreFocus(fe, newFocusedWindow);
                         }
                         break;
                     }
