diff -r 27e746e6f3fe src/share/classes/java/awt/TextComponent.java
--- a/src/share/classes/java/awt/TextComponent.java	Sun Feb 12 21:09:09 2012 +0000
+++ b/src/share/classes/java/awt/TextComponent.java	Mon Feb 13 01:42:20 2012 +0000
@@ -599,7 +599,7 @@
      * @since 1.4
      */
     public synchronized TextListener[] getTextListeners() {
-        return (TextListener[])(getListeners(TextListener.class));
+        return getListeners(TextListener.class);
     }
 
     /**
