diff -r 0678af55d3db src/share/classes/java/net/URLStreamHandler.java
--- a/src/share/classes/java/net/URLStreamHandler.java	Tue Jun 05 17:11:26 2012 +0800
+++ b/src/share/classes/java/net/URLStreamHandler.java	Tue Jun 05 19:59:06 2012 +0100
@@ -474,7 +474,7 @@
      */
     protected String toExternalForm(URL u) {
 
-        // pre-compute length of StringBuffer
+        // pre-compute length of StringBuilder
         int len = u.getProtocol().length() + 1;
         if (u.getAuthority() != null && u.getAuthority().length() > 0)
             len += 2 + u.getAuthority().length();
@@ -487,7 +487,7 @@
         if (u.getRef() != null)
             len += 1 + u.getRef().length();
 
-        StringBuffer result = new StringBuffer(len);
+        StringBuilder result = new StringBuilder(len);
         result.append(u.getProtocol());
         result.append(":");
         if (u.getAuthority() != null && u.getAuthority().length() > 0) {
