diff -r 7baa22e6a6b3 src/share/classes/javax/swing/plaf/synth/SynthLookAndFeel.java
--- a/src/share/classes/javax/swing/plaf/synth/SynthLookAndFeel.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/plaf/synth/SynthLookAndFeel.java	Mon Jun 04 20:03:50 2012 +0100
@@ -64,6 +64,9 @@
  * @author Scott Violet
  */
 public class SynthLookAndFeel extends BasicLookAndFeel {
+
+    private static final long serialVersionUID = 1803572249648355104L;
+
     /**
      * Used in a handful of places where we need an empty Insets.
      */
