diff -r 1980be18d0f8 src/share/classes/java/util/InvalidPropertiesFormatException.java
--- a/src/share/classes/java/util/InvalidPropertiesFormatException.java	Sun Apr 22 21:22:17 2012 +0100
+++ b/src/share/classes/java/util/InvalidPropertiesFormatException.java	Tue Apr 24 01:15:51 2012 +0100
@@ -44,6 +44,9 @@
  */
 
 public class InvalidPropertiesFormatException extends IOException {
+
+    private static final long serialVersionUID = 7763056076009360219L;
+
     /**
      * Constructs an InvalidPropertiesFormatException with the specified
      * cause.
diff -r dfe5617c18b4 src/share/classes/java/util/regex/Pattern.java
--- a/src/share/classes/java/util/regex/Pattern.java	Wed Jun 20 22:40:32 2012 -0400
+++ b/src/share/classes/java/util/regex/Pattern.java	Mon Jun 25 13:36:20 2012 +0100
@@ -2026,6 +2026,7 @@
         }
     }
 
+    @SuppressWarnings("fallthrough")
     /**
      * Parsing of sequences between alternations.
      */
@@ -2140,6 +2141,7 @@
         return head;
     }
 
+    @SuppressWarnings("fallthrough")
     /**
      * Parse and add a new Single or Slice.
      */
@@ -2978,6 +2980,7 @@
         return head;
     }
 
+    @SuppressWarnings("fallthrough")
     /**
      * Parses inlined match flags and set them appropriately.
      */
@@ -3019,6 +3022,7 @@
         }
     }
 
+    @SuppressWarnings("fallthrough")
     /**
      * Parses the second part of inlined match flags and turns off
      * flags appropriately.
diff -r 1980be18d0f8 src/share/classes/java/util/zip/GZIPInputStream.java
--- a/src/share/classes/java/util/zip/GZIPInputStream.java	Sun Apr 22 21:22:17 2012 +0100
+++ b/src/share/classes/java/util/zip/GZIPInputStream.java	Mon Apr 23 19:36:16 2012 +0100
@@ -252,7 +252,7 @@
      */
     private int readUShort(InputStream in) throws IOException {
         int b = readUByte(in);
-        return ((int)readUByte(in) << 8) | b;
+        return (readUByte(in) << 8) | b;
     }
 
     /*
