diff -r 0678af55d3db src/share/classes/javax/swing/plaf/nimbus/State.java
--- a/src/share/classes/javax/swing/plaf/nimbus/State.java	Tue Jun 05 17:11:26 2012 +0800
+++ b/src/share/classes/javax/swing/plaf/nimbus/State.java	Tue Jun 05 19:59:26 2012 +0100
@@ -179,7 +179,7 @@
         }
 
         private static String toString(int state) {
-            StringBuffer buffer = new StringBuffer();
+            StringBuilder buffer = new StringBuilder();
             if ((state & SynthConstants.DEFAULT) == SynthConstants.DEFAULT) {
                 buffer.append("Default");
             }
