diff -r 134d9f4f9498 src/share/classes/com/sun/media/sound/UlawCodec.java
--- a/src/share/classes/com/sun/media/sound/UlawCodec.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/com/sun/media/sound/UlawCodec.java	Mon Mar 12 23:16:12 2012 +0000
@@ -302,7 +302,7 @@
 
             // set the AudioInputStream length in frames if we know it
             if (stream instanceof AudioInputStream) {
-                frameLength = ((AudioInputStream)stream).getFrameLength();
+                frameLength = stream.getFrameLength();
             }
             // set framePos to zero
             framePos = 0;
@@ -407,8 +407,8 @@
                     return readCount;
                 }
                 for (i = off; i < (off + (readCount*2)); i+=2) {
-                    b[i]        = (byte)tabByte1[b[readOffset] & 0xFF];
-                    b[i+1]      = (byte)tabByte2[b[readOffset] & 0xFF];
+                    b[i]        = tabByte1[b[readOffset] & 0xFF];
+                    b[i+1]      = tabByte2[b[readOffset] & 0xFF];
                     readOffset++;
                 }
                 return (i - off);
