diff -r 134d9f4f9498 src/share/classes/com/sun/media/sound/WaveFileWriter.java
--- a/src/share/classes/com/sun/media/sound/WaveFileWriter.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/com/sun/media/sound/WaveFileWriter.java	Mon Mar 12 23:16:12 2012 +0000
@@ -248,17 +248,17 @@
 
             if (maxLength>0) {
                 if( bytesRead < maxLength ) {
-                    out.write( buffer, 0, (int)bytesRead );
+                    out.write( buffer, 0, bytesRead );
                     bytesWritten += bytesRead;
                     maxLength -= bytesRead;
                 } else {
-                    out.write( buffer, 0, (int)maxLength );
+                    out.write( buffer, 0, maxLength );
                     bytesWritten += maxLength;
                     maxLength = 0;
                     break;
                 }
             } else {
-                out.write( buffer, 0, (int)bytesRead );
+                out.write( buffer, 0, bytesRead );
                 bytesWritten += bytesRead;
             }
         }
@@ -280,7 +280,7 @@
         short channels         = (short) audioFormat.getChannels();
         short sampleSizeInBits = (short) audioFormat.getSampleSizeInBits();
         int sampleRate         = (int) audioFormat.getSampleRate();
-        int frameSizeInBytes   = (int) audioFormat.getFrameSize();
+        int frameSizeInBytes   = audioFormat.getFrameSize();
         int frameRate              = (int) audioFormat.getFrameRate();
         int avgBytesPerSec     = channels * sampleSizeInBits * sampleRate / 8;;
         short blockAlign       = (short) ((sampleSizeInBits / 8) * channels);
