diff -r 7baa22e6a6b3 src/share/classes/javax/swing/JPanel.java
--- a/src/share/classes/javax/swing/JPanel.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/JPanel.java	Mon Jun 04 20:01:49 2012 +0100
@@ -64,6 +64,9 @@
  */
 public class JPanel extends JComponent implements Accessible
 {
+
+    private static final long serialVersionUID = 6720508106274062884L;
+
     /**
      * @see #getUIClassID
      * @see #readObject
