diff -r 7baa22e6a6b3 src/share/classes/javax/swing/undo/AbstractUndoableEdit.java
--- a/src/share/classes/javax/swing/undo/AbstractUndoableEdit.java	Fri Jun 01 00:05:21 2012 -0700
+++ b/src/share/classes/javax/swing/undo/AbstractUndoableEdit.java	Mon Jun 04 20:04:34 2012 +0100
@@ -37,6 +37,8 @@
  */
 public class AbstractUndoableEdit implements UndoableEdit, Serializable {
 
+    private static final long serialVersionUID = 580150227676302096L;
+
     /**
      * String returned by <code>getUndoPresentationName</code>;
      * as of Java 2 platform v1.3.1 this field is no longer used. This value
