diff -r 134d9f4f9498 src/share/classes/com/sun/media/sound/RealTimeSequencer.java
--- a/src/share/classes/com/sun/media/sound/RealTimeSequencer.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/com/sun/media/sound/RealTimeSequencer.java	Mon Mar 12 23:16:11 2012 +0000
@@ -392,7 +392,7 @@
             // last resort: return a standard tempo: 120bpm
             return (float) MidiUtils.DEFAULT_TEMPO_MPQ;
         }
-        return (float)getDataPump().getTempoMPQ();
+        return getDataPump().getTempoMPQ();
     }
 
 
