diff -r 27e746e6f3fe src/share/classes/java/awt/Button.java
--- a/src/share/classes/java/awt/Button.java	Sun Feb 12 21:09:09 2012 +0000
+++ b/src/share/classes/java/awt/Button.java	Mon Feb 13 01:39:52 2012 +0000
@@ -300,7 +300,7 @@
      * @since 1.4
      */
     public synchronized ActionListener[] getActionListeners() {
-        return (ActionListener[]) (getListeners(ActionListener.class));
+        return getListeners(ActionListener.class);
     }
 
     /**
