diff -r 134d9f4f9498 src/share/classes/java/awt/MenuComponent.java
--- a/src/share/classes/java/awt/MenuComponent.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/java/awt/MenuComponent.java	Mon Mar 12 23:16:23 2012 +0000
@@ -288,7 +288,7 @@
     public void setFont(Font f) {
         font = f;
         //Fixed 6312943: NullPointerException in method MenuComponent.setFont(Font)
-        MenuComponentPeer peer = (MenuComponentPeer)this.peer;
+        MenuComponentPeer peer = this.peer;
         if (peer != null) {
             peer.setFont(f);
         }
@@ -301,7 +301,7 @@
      */
     public void removeNotify() {
         synchronized (getTreeLock()) {
-            MenuComponentPeer p = (MenuComponentPeer)this.peer;
+            MenuComponentPeer p = this.peer;
             if (p != null) {
                 Toolkit.getEventQueue().removeSourceEvents(this, true);
                 this.peer = null;
