diff -r 134d9f4f9498 src/share/classes/java/awt/image/ComponentSampleModel.java
--- a/src/share/classes/java/awt/image/ComponentSampleModel.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/java/awt/image/ComponentSampleModel.java	Mon Mar 12 23:16:19 2012 +0000
@@ -147,7 +147,7 @@
         this.dataType = dataType;
         this.pixelStride = pixelStride;
         this.scanlineStride  = scanlineStride;
-        this.bandOffsets = (int[])bandOffsets.clone();
+        this.bandOffsets = bandOffsets.clone();
         numBands = bandOffsets.length;
         if (pixelStride < 0) {
             throw new IllegalArgumentException("Pixel stride must be >= 0");
@@ -210,8 +210,8 @@
         this.dataType = dataType;
         this.pixelStride = pixelStride;
         this.scanlineStride  = scanlineStride;
-        this.bandOffsets = (int[])bandOffsets.clone();
-        this.bankIndices = (int[]) bankIndices.clone();
+        this.bandOffsets = bandOffsets.clone();
+        this.bankIndices = bankIndices.clone();
         if (pixelStride < 0) {
             throw new IllegalArgumentException("Pixel stride must be >= 0");
         }
@@ -496,14 +496,14 @@
      *  @return the bank indices for all bands.
      */
     public final int [] getBankIndices() {
-        return (int[]) bankIndices.clone();
+        return bankIndices.clone();
     }
 
     /** Returns the band offset for all bands.
      *  @return the band offsets for all bands.
      */
     public final int [] getBandOffsets() {
-        return (int[])bandOffsets.clone();
+        return bandOffsets.clone();
     }
 
     /** Returns the scanline stride of this ComponentSampleModel.
