diff -r 134d9f4f9498 src/share/classes/java/awt/image/DataBufferShort.java
--- a/src/share/classes/java/awt/image/DataBufferShort.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/share/classes/java/awt/image/DataBufferShort.java	Mon Mar 12 23:16:21 2012 +0000
@@ -153,7 +153,7 @@
      */
     public DataBufferShort(short dataArray[][], int size) {
         super(UNTRACKABLE, TYPE_SHORT, size, dataArray.length);
-        bankdata = (short[][]) dataArray.clone();
+        bankdata = dataArray.clone();
         data = bankdata[0];
     }
 
@@ -178,7 +178,7 @@
      */
     public DataBufferShort(short dataArray[][], int size, int offsets[]) {
         super(UNTRACKABLE, TYPE_SHORT, size, dataArray.length, offsets);
-        bankdata = (short[][]) dataArray.clone();
+        bankdata = dataArray.clone();
         data = bankdata[0];
     }
 
@@ -225,7 +225,7 @@
      */
     public short[][] getBankData() {
         theTrackable.setUntrackable();
-        return (short[][]) bankdata.clone();
+        return bankdata.clone();
     }
 
     /**
