diff -r 134d9f4f9498 src/solaris/classes/sun/awt/X11/XDropTargetRegistry.java
--- a/src/solaris/classes/sun/awt/X11/XDropTargetRegistry.java	Mon Mar 12 11:30:28 2012 +0000
+++ b/src/solaris/classes/sun/awt/X11/XDropTargetRegistry.java	Mon Mar 12 23:16:51 2012 +0000
@@ -329,7 +329,7 @@
         boolean isXEmbedServer = false;
         synchronized (this) {
             EmbeddedDropSiteEntry entry =
-                (EmbeddedDropSiteEntry)embeddedDropSiteRegistry.get(lToplevel);
+                embeddedDropSiteRegistry.get(lToplevel);
             if (entry == null) {
                 return;
             }
@@ -430,8 +430,7 @@
         Long lToplevel = Long.valueOf(toplevel);
         EmbeddedDropSiteEntry entry = null;
         synchronized (this) {
-            entry =
-                (EmbeddedDropSiteEntry)embeddedDropSiteRegistry.get(lToplevel);
+            entry = embeddedDropSiteRegistry.get(lToplevel);
             if (entry == null) {
                 if (peer != null) {
                     // Toplevel is an XEmbed server within this VM.
@@ -495,8 +494,7 @@
         Long lToplevel = Long.valueOf(toplevel);
         EmbeddedDropSiteEntry entry = null;
         synchronized (this) {
-            entry =
-                (EmbeddedDropSiteEntry)embeddedDropSiteRegistry.get(lToplevel);
+            entry = embeddedDropSiteRegistry.get(lToplevel);
             if (entry == null) {
                 return;
             }
@@ -526,8 +524,7 @@
      */
     public long getEmbeddedDropSite(long embedder, int x, int y) {
         Long lToplevel = Long.valueOf(embedder);
-        EmbeddedDropSiteEntry entry =
-            (EmbeddedDropSiteEntry)embeddedDropSiteRegistry.get(lToplevel);
+        EmbeddedDropSiteEntry entry = embeddedDropSiteRegistry.get(lToplevel);
         if (entry == null) {
             return 0;
         }
